k2/csrc/arcsort.cc

```diff
@@ -0,0 +1,67 @@
+// k2/csrc/arcsort.cc
+
+// Copyright (c)  2020  Xiaomi Corporation (author: Haowen Qiu)
+
+// See ../../LICENSE for clarification regarding multiple authors
+
+#include "k2/csrc/arcsort.h"
+
+#include <algorithm>
+#include <numeric>
+#include <utility>
+#include <vector>
+
+#include "glog/logging.h"
+#include "k2/csrc/fsa.h"
+
+namespace k2 {
+void ArcSorter::GetSizes(Array2Size<int32_t> *fsa_size) {
+  CHECK_NOTNULL(fsa_size);
+  fsa_size->size1 = fsa_in_.size1;
+  fsa_size->size2 = fsa_in_.size2;
+}
+
+void ArcSorter::GetOutput(Fsa *fsa_out, int32_t *arc_map /*= nullptr*/) {
+  CHECK_NOTNULL(fsa_out);
+  CHECK_EQ(fsa_out->size1, fsa_in_.size1);
+  CHECK_EQ(fsa_out->size2, fsa_in_.size2);
+
+  using ArcWithIndex = std::pair<Arc, int32_t>;
+  std::vector<int32_t> indexes(fsa_in_.size2);  // arc index mapping
```

Comment:
Need to be clearer here, it's a mapping from what to what? 
E.g. if indexes[i] == j, is index i an arc-index into fsa_out at entry, and index j an arc-index into fsa_out at exit?
Or vice versa?